typespec-extension/src/main/java/com/azure/autorest/TypeSpecPlugin.java

```diff
@@ -57,6 +58,15 @@ protected void writeClientModels(Client client, JavaPackage javaPackage, JavaSet
                 .filter(ModelUtil::isGeneratingModel)
                 .forEach(model -> javaPackage.addModel(model.getPackage(), model.getName(), model));
 
+        // JsonMergePatchHelper
+        boolean hasJsonMergePatchModelToGenerate = client.getModels().stream()
+                .filter(ModelUtil::isGeneratingModel)
+                .filter(ClientModelUtil::isJsonMergePatchModel)
+                .findFirst().isPresent();
+//        if (hasJsonMergePatchModelToGenerate) {
+//            javaPackage.addJsonMergePatchHelper(client.getModels());
+//        }
```

Comment:
recommend to filter the models (as you already done here), and only send these to the template.

and template no longer need to do the filter again.

Reply:
sure, will pass in the filtered models to addJsonMergePatchHelper(), but 'm not sure if we want to delete the checking logic in template, as template should not assume all the passed-in models are json merge patch model, it's safer to add  the check in template as well? What do you think?

Reply:
Of course don't delete the check logic. It just becomes "if list is not empty"

Reply:
I mean  the logic on checking whether model is json merge patch model. Do we want to keep it for safety?

Reply:
No. For safety you should do null check on every single string usage, do you?

Reply:
ok, fine.